### ISSUE TITLE
Warn user if something is missing during phyloseq-class creation

### ISSUE THREAD

**Issue by vmikk** [joey711/phyloseq#685]
Hello!
It would be really nice to add some warnings during phyloseq-class creation if something is missing in some of the slots (e.g., not all samples are in `sample_data`).
Here is a simple example:
```
library(phyloseq)

# Dummy OTU table
otumat <- matrix(sample(1:100, 100, replace = TRUE), nrow = 10, ncol = 10)
rownames(otumat) <- paste0("OTU", 1:nrow(otumat))
colnames(otumat) <- paste0("Sample", 1:ncol(otumat))

# Create phyloseq object
physeq <- phyloseq(otu_table(otumat, taxa_are_rows = TRUE))

# Sample metadata
sampmat <- data.frame(Location = sample(LETTERS[1:4], size = 5, replace = TRUE))
row.names(sampmat) = paste0("Sample", 1:nrow(sampmat))

# Add sample metadata to physeq
physeq <- merge_phyloseq(physeq, sample_data(sampmat))

# or recreate physeq from scratch
physeq2 <- phyloseq(otu_table(otumat, taxa_are_rows = TRUE), sample_data(sampmat))
```
Everything seems fine. But when you will call `physeq` or `physeq2` you will notice that half of the samples were **silently** dropped because they had no metadata.

So maybe it has sense to make `phyloseq` more verbose and issue a warning in such case?

**Comment by joey711** [comment]
I agree. This is a good idea, and not that difficult to add. I think what I'll do is add a `verbose` option to the phyloseq class constructor, and set it to `TRUE` by default. Differences in dimensions for each input component and the final dimensions will be reported as (length capped?) messages to standard out.

Anything I miss? I'll keep this open as a feature request until it's in the package.

Thanks!

**Comment by vmikk** [comment]
Hello @joey711! Thank you for the feedback and you your time!
I think `verbose` option would be superb!
